app/src/main/java/com/github/se/polyfit/ui/screen/GraphScreen.kt

```diff
@@ -0,0 +1,126 @@
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.SpaceAround) {
+                      if (!isTestEnvironment) {
+                        LineChart(
+                            modifier = Modifier.fillMaxSize().testTag("LineChart"),
+                            lineChartData = lineChartData(DataToPoints(), DateList()))
+                      } else {
+                        Spacer(modifier = Modifier.fillMaxSize().testTag("LineChartSpacer"))
+                      }
+                    }
+              }
+              Spacer(modifier = Modifier.height(5.dp))
+              Row(
+                  modifier = Modifier.fillMaxWidth().padding(5.dp),
+                  horizontalArrangement = Arrangement.SpaceBetween) {
+                    searchText?.let {
+                      OutlinedTextField(
+                          value = it,
+                          onValueChange = viewModel::onSearchTextChanges,
+                          modifier = Modifier.fillMaxWidth(0.35f).testTag("GraphScreenSearchBar"),
+                          placeholder = { Text(text = "Search") },
+                          shape = MaterialTheme.shapes.large)
+                    }
+                    DropDownMenu(
+                        isExpanded,
+                        SortPoints.toList(),
+                        sortedBy,
+                        onItemSelect = { sortAttributeName ->
+                          val sortAttribute = SortPoints.valueOf(sortAttributeName)
+                          viewModel.updateSort(sortAttribute)
+                        })
+                  }
+              Spacer(modifier = Modifier.height(5.dp))
+              HorizontalDivider(thickness = 1.dp, color = MaterialTheme.colorScheme.inversePrimary)
+              LazyColumn(modifier = Modifier.fillMaxWidth().weight(1f).testTag("ElementsList")) {
+                graphData?.let {
+                  items(it) { data ->
+                    Spacer(modifier = Modifier.height(5.dp))
+                    Row(
+                        modifier = Modifier.fillMaxWidth().testTag("ElementsRow"),
+                        horizontalArrangement = Arrangement.SpaceBetween) {
+                          Column(
+                              horizontalAlignment = Alignment.Start,
+                              modifier = Modifier.padding(start = 4.dp)) {
+                                Text(
+                                    text = "${data.kCal} kCal", modifier = Modifier.testTag("kcal"))
+                                Text(
+                                    text = "${data.weight} lbs",
+                                    modifier = Modifier.testTag("weight"))
+                              }
+                          Text(
+                              modifier = Modifier.padding(end = 4.dp).testTag("Date"),
+                              text = "${data.day} ${data.month}")
```

Comment:
same question about having string values stored external

Reply:
the String.xml is usually intended for static strings which isn't the case here

Reply:
I don't think it is unique for static strings: https://developer.android.com/develop/ui/compose/resources#strings